### ISSUE TITLE
Photoshop CS4 plugin does not write into file

### ISSUE THREAD

**Issue by alchumie** [webmproject/WebPShop#39]
The plug in for WebP is installed and is recognized.  I can open a WebP file, but when I save it gives me the interface but writes nothing to the file.  I get a webP file with 0 bytes.  Anyone have a clue what's going on?

**Comment by y-guyon** [comment]
Unfortunately, older versions of Photoshop are not supported. Please see [issue #17](https://github.com/webmproject/WebPShop/issues/17) for more details. You can use [Telegraphics' plug-in](http://www.telegraphics.com.au/sw/product/WebPFormat) as an alternative.

**Comment by alchumie** [comment]
Well, I can confirm that the Telegraphics plug in for Photoshop CS4 works fine and creates a file.  However when the file is uploaded to a web server, the server is unable to serve it up in HTML pages.  Here is a link to the webp image https://www.dogsindanger.com/doggie/images/dogsindangerDog_mobile.webp.  What's the deal?

**Comment by y-guyon** [comment]
The file is broken. Output of [`webpinfo -diag`](https://chromium.googlesource.com/webm/libwebp/+/v0.6.1-rc2/examples/webpinfo.c):
```
RIFF HEADER:
  File size:  13074
Error: Truncated data detected when parsing RIFF payload.
```
You can try contacting Telegraphics but I believe their plugin is no longer under active development.

There is another plug-in developed by fnordware that you might try too: [AdobeWebM](https://github.com/fnordware/AdobeWebM).

**Comment by alchumie** [comment]
Yes, useless to contact Telegraphics as you mentioned not under development. AdobeWenM Photoshop plug in download link useless, nothing there.  Probably not in usable shape.  Wow, I never though it would be so hard to create WebP files. And everyone is recommending it for mobile.

**Comment by y-guyon** [comment]
Try directly from http://www.fnordware.com/WebM/. It works for me.

Alternatives:

- A recent Photoshop version
- [Gimp](https://www.gimp.org/)
- Command line tools ([ImageMagick](https://imagemagick.org/script/convert.php), [libwebp binaries](https://developers.google.com/speed/webp/docs/precompiled))
- Online converters

**Comment by alchumie** [comment]
Yes it works but the link is for Premiere, not Photoshop.  Sadly I don't have Premiere. Thanks again!

**Comment by y-guyon** [comment]
"Right now there's a working WebM plug-in for Premiere Pro, along with a beta WebP plug-in for Photoshop." from the GitHub repository, so it is probably bundled with it.

**Comment by alchumie** [comment]
Sorry but it does not look like it.  During installation it looked for Premiere, which I do not have.
